File: reto5/reto5.ts
type Shoe = {
  type: "I" | "R";
  size: number;
};

function organizeShoes(shoes: Shoe[]): number[] {
  const pairs: Record<number, { I: number; R: number }> = {};

  for (const shoe of shoes) {
    pairs[shoe.size] = pairs[shoe.size] || { I: 0, R: 0 };
    pairs[shoe.size][shoe.type]++;
  }

  const result: number[] = [];
  for (const size in pairs) {
    const count = Math.min(pairs[size].I, pairs[size].R);
    for (let i = 0; i < count; i++) result.push(Number(size));
  }

  return result;
}

File: reto5/reto5.py
def organizeShoes(shoes):
    pairs = {}

    for shoe in shoes:
        pairs[shoe['size']] = pairs.get(shoe['size'], {'I': 0, 'R': 0})
        pairs[shoe['size']][shoe['type']] += 1

    result = []
    for size in pairs:
        count = min(pairs[size]['I'], pairs[size]['R'])
        for i in range(count):
            result.append(size)

    return result
